src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -543,7 +543,14 @@ await EnqueueWorkItemAsync(projectChanges.OldProject.GetDocument(changedDocument
             private async Task EnqueueWorkItemAsync(Document oldDocument, Document newDocument)
             {
                 var differenceService = newDocument.GetLanguageService<IDocumentDifferenceService>();
-                if (differenceService != null)
+
+                if (differenceService == null)
+                {
+                    // For languages that don't use a Roslyn syntax tree, they don't export a document difference service.
+                    // The whole document should be considered as changed in that case.
+                    await EnqueueWorkItemAsync(newDocument, InvocationReasons.DocumentChanged).ConfigureAwait(false);
```

Comment:
DocumentChanged will cause p2p propagation. which project is this document belong to? do you want all other documents to be re-analyzed (due to semantic changes) and propagate it up to dependency chain? if so, DocumentChanged seems right one. otherwise, you should use things like SyntaxChanged to make only that document to be re-analyzed.